examples/README

```diff
@@ -0,0 +1,8 @@
+Build all examples using "make" or "make all".
```

Comment:
Should we mention about adding `oshcc` and `oshrun` in the PATH? Also, adding the library path.

Reply:
Yeah, we probably should mention the bare minimum requirements to run these examples.  Something like:

> Requirements to build and run the examples:
> * Have the SOS installation "bin" directory to your PATH.
> * Have a compatible process launcher, like Hydra version 3.2, on your path.
> * The hello world example can be run with 4 processes like so:
>           oshrun -n 4 ./hello

Although it might be better to keep this `examples` directory self-contained, perhaps we could also provide a link to the new wiki page "for more detailed information":
https://github.com/Sandia-OpenSHMEM/SOS/wiki/Getting-Started-Guide
Specifically, we could directly link to the ["Building"](https://github.com/kholland-intel/SOS/wiki/Getting-Started-Guide#compiling-an-openshmem-program) and ["Running"](https://github.com/kholland-intel/SOS/wiki/Getting-Started-Guide#running-an-openshmem-program) OpenSHMEM programs sections.

I put a placeholder document on the official SOS wiki for now - feel free to update it now @kholland-intel, it looks ready to me and we can fine tune it later if needed ;)

Reply:
Done.